dev/update_changelog.py

```diff
@@ -0,0 +1,290 @@
+            elif pr_type == "break":
+                breaking_changes = True
+                insert_content_index = content.find(
+                    "### Incompatible changes", unreleased_index + 1
+                )
+            elif pr_type in {"ci", "fix", "refactor"}:
+                insert_content_index = content.find(
+                    "### Other changes", unreleased_index + 1
+                )
+            else:
+                unknown_changes = True
+                insert_content_index = unreleased_index
+
+            pr_reference = _format_pr_reference(
+                pr_info.title, pr_info.number, pr_info.html_url
+            )
+
+            content = _insert_entry_no_desc(
+                content,
+                pr_reference,
+                insert_content_index,
+            )
+
+            next_header_index = content.find("## ", unreleased_index + 1)
+            next_header_index = (
+                next_header_index if next_header_index != -1 else len(content)
+            )
+
+        if unknown_changes:
+            content = _insert_entry_no_desc(
+                content,
+                "### Unknown changes",
+                unreleased_index,
+            )
+
+        if not breaking_changes:
+            content = _insert_entry_no_desc(
+                content,
+                "None",
+                content.find("### Incompatible changes", unreleased_index + 1),
+            )
+
+        # Finalize content update
+        file.seek(0)
+        file.write(content)
+        file.truncate()
+
+
+def _insert_entry_no_desc(
+    content: str, pr_reference: str, unreleased_index: int
+) -> str:
+    """Insert a changelog entry for a pull request with no specific description."""
+    insert_index = content.find("\n", unreleased_index) + 1
+    content = (
+        content[:insert_index] + "\n" + pr_reference + "\n" + content[insert_index:]
+    )
+    return content
+
+
+def _bump_minor_version(tag: Tag) -> Optional[str]:
+    """Bump the minor version of the tag."""
+    match = re.match(r"v(\d+)\.(\d+)\.(\d+)", tag.name)
+    if match is None:
+        return None
+    major, minor, _ = [int(x) for x in match.groups()]
+    # Increment the minor version and reset patch version
+    new_version = f"v{major}.{minor + 1}.0"
+    return new_version
+
+
+def main() -> None:
+    """Update changelog using the descriptions of PRs since the latest tag."""
+    # Initialize GitHub Client with provided token (as argument)
+    gh_api = Github(argv[1])
+    repo, latest_tag = _get_latest_tag(gh_api)
+    if not latest_tag:
+        print("No tags found in the repository.")
+        return
+
+    shortlog, prs = _get_pull_requests_since_tag(repo, latest_tag)
+    _update_changelog(prs)
+
+    new_version = _bump_minor_version(latest_tag)
+    if not new_version:
+        print("Wrong tag format.")
+        return
+
+    _add_shorlog(new_version, shortlog)
+
+    print("Changelog updated succesfully.")
```

Comment:
Maybe we should always print this? If there's no `## Unreleased` tag, we will see:

```
Unreleased header not found in the changelog.
Changelog updated succesfully.
```